object/object_test.go

```diff
@@ -0,0 +1,441 @@
+						{
+							"type": "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			want: map[string]interface{}{
+				"conditions": []map[string]interface{}{
+					{
+						"type": "WaitingForPlayer",
+					},
+				},
+			},
+		},
+		{
+			name: "multiple status conditions",
+			obj: map[string]interface{}{
+				"apiVersion": "v1alpha1",
+				"kind":       "Game",
+				"metadata": map[string]interface{}{
+					"creationTimestamp": nil,
+				},
+				"spec": map[string]interface{}{},
+				"status": map[string]interface{}{
+					"conditions": []map[string]interface{}{
+						{
+							"type": "WaitingForPlayer",
+						},
+						{
+							"type": "Online",
+						},
+					},
+				},
+			},
+			want: map[string]interface{}{
+				"conditions": []map[string]interface{}{
+					{
+						"type": "WaitingForPlayer",
+					},
+					{
+						"type": "Online",
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := GetObjectStatus(tc.obj)
+			if tc.wantErr {
+				assert.Error(t, err)
+				assert.Nil(t, got)
+				return
+			}
+			assert.NoError(t, err)
+			assert.Equal(t, tc.want, got, "result")
+		})
+	}
+}
+
+func TestStatusChanged(t *testing.T) {
+	// Create a scheme with testdata scheme info.
+	scheme := runtime.NewScheme()
+	assert.Nil(t, tdv1alpha1.AddToScheme(scheme))
+
+	cases := []struct {
+		name    string
+		oldo    runtime.Object
+		newo    runtime.Object
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "no status",
+			oldo: &tdv1alpha1.Game{},
+			newo: &tdv1alpha1.Game{},
+			want: false,
+		},
+		{
+			name: "no change",
+			oldo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			newo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			want: false,
+		},
+		{
+			name: "status added",
+			oldo: &tdv1alpha1.Game{},
+			newo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "Booting",
+						},
+					},
+				},
+			},
+			want: true,
+		},
+		{
+			name: "status removed",
+			oldo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "SelfDestructing",
+						},
+					},
+				},
+			},
+			newo: &tdv1alpha1.Game{},
+			want: true,
+		},
+		{
+			name: "add additional",
+			oldo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+					},
+				},
+			},
+			newo: &tdv1alpha1.Game{
+				Status: tdv1alpha1.GameStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type: "WaitingForPlayer",
+						},
+						{
+							Type: "InstallingUpdate",
+						},
+					},
+				},
+			},
+			want: true,
+		},
```

Comment:
We can add a few failure cases for this, maybe passing an object without any status?
But I guess those things are being tested separately in functions like `GetObjectStatus`. Maybe we can just remove the `wantErr` from the test cases?

Reply:
Since we're passing in a type, it's not possible to provide input that will trigger an error in one of the deeper functions that accept an interface{}.  i.e. Status will always be valid, even if empty.

Instead, I've removed `wantErr` as the error conditions are tested separately.